app/src/main/java/de/xikolo/App.kt

```diff
@@ -69,6 +77,13 @@ class App : Application() {
         configureRealm()
         configureDefaultSettings()
         configureWebView()
+
+        val recentCoursesObserver = Observer<String> {
+            updateShortcuts()
+        }
+        val recentCourses = RecentCoursesStorage()
+        //who is the owner?
+        recentCourses.coursesLive.observe(this, recentCoursesObserver)
```

Comment:
There#s an error, that this is not the correct owner. Can somebody help me with that?

Reply:
Which interface is expected for the owner? Find that out and look which classes implement that to choose the right owner and right place for that code. Also, this code should be moved to a method.